### ISSUE TITLE
[Train] Custom checkpoint directory name for Trainer like Tuner does

### ISSUE THREAD

**Issue by daviddwlee84** [ray-project/ray#41817]
### Description

I want to customize the checkpoint directory name that Trainer creates instead of using a default name like `storage_path/experiment_name/TrainerClassName_unique_id_00000_0_datetime`

We have `trial_dirname_creator` in [`ray.tune.TuneConfig`](https://docs.ray.io/en/latest/tune/api/doc/ray.tune.TuneConfig.html).
And Trainer creates Tuner during `fit()`. But not passing TuneConfig to it.

If passing a TuneConfig as an argument of `Trainer.fit()` confuses users, maybe adding an option in Trainer's RunConfig can be a good place (since we set the experiment name and storage path there). But either way works.

https://github.com/ray-project/ray/blob/171b828177420b2530d9fbe07f6914cdbf6ee0ba/python/ray/train/base_trainer.py#L578-L617

https://github.com/ray-project/ray/issues/10648

### Use case

I think it is more like a "manual Tuner" scenario.
Let's say using the same model parameter for an `experiment_name`, but change a different dataset date range, I want this information to show on the checkpoint directory name like `storage_path/experiment_name/{dataset_date_range}_unique_id`.
And if I want to have a different learning rate at the same time, I can add to the path `storage_path/experiment_name/{dataset_date_range}_{learning_rate}_unique_id`.

**Comment by matthewdeng** [comment]
Ah this seems like a duplicate of https://github.com/ray-project/ray/issues/40531

@justinvyu do you think it would make sense to start with a simple experimental config?

**Comment by daviddwlee84** [comment]
Oh... right. I missed that one. 😂

**Comment by justinvyu** [comment]
@daviddwlee84 Is this workaround suitable for now?

https://github.com/ray-project/ray/issues/40531#issuecomment-1828536267

Also open to adding this experimental config if you prefer to stick with working with the `Trainer` only.

**Comment by daviddwlee84** [comment]
Yes, this is suitable.
Even though it is not very intuitive for users who haven't used Tuner before, since it is hard to find in the Trainer's document.

Both ways are reasonable to me.

If the concept that "a single Train == a tuner Trial round" exists, I think using TuneConfig (with or without wrapped with Tuner) would be more elegant for me.
Otherwise, if users don't need to know Tuner is underneath Trainer, maybe making this a part of Trainer's config would be more friendly.

**Comment by daturkel** [comment]
This issue is still creating a problem for our team as we try to automate more of our training flow. I would love if this name is exposed via some configuration. For the time being, we're using Weights and Biases artifacts to get around ray's built-in artifact syncing.

**Comment by sam-h-bean** [comment]
Please expose this so that we can deterministically find the latest checkpoint more easily in slurm! This lack of functionality makes it relaly hard to delegate restarting jobs to slurm since we can not reliably pass back in the `resume_from_checkpoint` directory once the process has exited and restarted.
